Apps/Common/EqualOrderOperators.C

```diff
@@ -54,17 +54,24 @@ static void DivGrad(Matrix& EM, const FiniteElement& fe,
 
 void EqualOrderOperators::Weak::Advection(Matrix& EM, const FiniteElement& fe,
                                           const Vec3& AC,
-                                          double scale, int basis)
+                                          double scale,
+                                          WeakOperators::ConvectionForm form,
+                                          int basis)
 {
   Matrix C(fe.basis(basis).size(), fe.basis(basis).size());
   size_t ncmp = EM.rows() / C.rows();
   for (size_t i = 1; i <= fe.basis(basis).size(); ++i) {
     for (size_t j = 1; j <= fe.basis(basis).size(); ++j) {
-      // Sum convection for each direction
-      for (size_t k = 1; k <= fe.grad(basis).cols(); ++k)
-        C(i,j) += AC[k-1]*fe.grad(basis)(j,k);
-
-      C(i,j) *= scale*fe.basis(basis)(i)*fe.detJxW;
+      if (form == WeakOperators::CONVECTIVE) {
+        // Sum convection for each direction
+        for (size_t k = 1; k <= fe.grad(basis).cols(); ++k)
+          C(i,j) += AC[k-1]*fe.grad(basis)(j,k);
+        C(i,j) *= scale*fe.basis(basis)(i)*fe.detJxW;
+      } else if (form == WeakOperators::CONSERVATIVE) {
+        for (size_t k = 1; k <= fe.grad(basis).cols(); ++k)
+          C(i,j) -= AC[k-1] * fe.grad(basis)(i,k);
+        C(i,j) *= fe.basis(basis)(j) * scale * fe.detJxW;
```

Comment:
Purely cosmetic, but in L72-73 you have space around the "*"-operators and different order of the factors compared to L68-69. Apart from that, the only difference between the two blocks are that "i" and "j" are swapped in the RHS, and -= instead of +=, but that would be easier to spot with similar coding style

Reply:
similarized.